pyfive/h5d.py

```diff
@@ -325,19 +330,29 @@ def _get_selection_via_chunks(self, args):
 
     @property
     def _fh(self):
-        """ 
+        """Return an open file handle to the parent file.
+
         When the parent file has been closed, we will need to reopen it
         to continue to access data. This facility is provided to support
         thread safe data access. However, now the file is open outside
         a context manager, the user is responsible for closing it,
         though it should get closed when the variable instance is
         garbage collected.
+
         """
-        if self.__fh.closed:
-            self.__fh = open(self._filename, 'rb')
-        return self.__fh
-    
-    
+        if self.posix:
+            # Posix: Open the file, without caching it.
+            return open(self._filename, 'rb')
+
+        # Not posix: Use the cached file if it's open, otherwise open
+        #            the file and cache it.
+        fh = self.__fh
+        if fh.closed:
```

Comment:
I am not sure whether this will work with S3 ... we probably need to do some testing around context managers and the like, but for the moment I am minded to approve it, then write the tests and deal with any issues after that.

Reply:
I think the [test script](https://github.com/NCAS-CMS/pyfive/pull/27#issue-2780434447) shows that it does work for S3. We can weave this script into V's moto framework.

Reply:
I am not sure it shows that it works for all the ways we might interact with it, but I think it shows enough for now.